### ISSUE TITLE
v4 navbar-fixed-top and modal-open padding-right content issue

### ISSUE THREAD

**Issue by iatek** [twbs/bootstrap#18366]
In v4, when using a fixed navbar, content gets pushed to the far left when a modal is opened. It occurs when the body content extends beyond the viewport (vertical scrollbar), and is caused by the `padding-right: 1617px;` that's being added the body.

Perhaps the _isBodyOverflowing && isModalOverflowing test in `modal.js` needs to compensate for the fixed navbar?

http://codeply.com/go/W1PphRIdfs

**Comment by Deilv** [comment]
I can't replicate, can you see how this is different?
http://codepen.io/deilv/pen/JYgoPV

**Comment by twbs-lmvtfy** [comment]
Hi @Deilv!

You appear to have posted a live example (http://codepen.io/deilv/pen/JYgoPV.html), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 127, column 7: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by Deilv** [comment]
Fixed missing type attr in http://codepen.io/deilv/pen/JYgoPV

**Comment by cvrebert** [comment]
I can reproduce within the iframe in the Codeply editor, but this doesn't reproduce when the page is top-level (i.e. not within an iframe).

**Comment by cvrebert** [comment]
This stops reproducing when I disable all the bullshit `::-webkit-scrollbar-` styles on the iframe.

**Comment by cvrebert** [comment]
@iatek Also FYI there seem to be some "srollbar" typos in the editor CSS.

**Comment by cvrebert** [comment]
Ok, I can reproduce with http://embed.plnkr.co/3lfONeCUI7IsWTehp9pZ/ after disabling autohiding scrollbars in OS X.

**Comment by cvrebert** [comment]
I suspect this is a duplicate of #14040, modulo the version.

**Comment by cvrebert** [comment]
[`_setScrollbar()`](https://github.com/twbs/bootstrap/blob/a23cf74340ef11ed7de81c2e105062b1e7f511e7/js/src/modal.js#L420)'s code seems wrong to me.
I think that
```js
`${bodyPadding + this._scrollbarWidth}px`
```
ought to use `this._originalBodyPadding` instead of `bodyPadding`.
And the compensation for `position:fixed` elements should be done on a per-element basis.

**Comment by Deilv** [comment]
@cvrebert Your example works as intended on FF/Edge. Could this be Safari/OSX specific?

**Comment by cvrebert** [comment]
Perhaps. I only tested on OS X; it reproduces there in Chrome, Firefox, and Safari.

**Comment by Deilv** [comment]
@cvrebert just for clarification, I am not talking about the slight shift of the navbar (I do see that and it's indeed a duplicate of #14040).
What I cannot see is the collapse of the content to the left. I did see it on the first codeply example, but nowhere else.

**Comment by Deilv** [comment]
The Planker example works properly on iOS/Safari, will also check on my mac later.

**Comment by cvrebert** [comment]
The distance of the shift of content to the left that I observed was small (~25px), not in the range of hundreds of px like OP said.

**Comment by Deilv** [comment]
If you read the issue description you will realize this isn't the case. On the codeply example the content collapsed all the way to the left, not just a few pixels. I believe it's a problem with codeply, not bootstrap. The minor shift is a known issue, irrelevant to this one. IMO closure is in order.

**Comment by cvrebert** [comment]
Closing as probable duplicate of #17399.

**Comment by cvrebert** [comment]
Opened #18373 to track the "content shifts only slightly to the left" bug.
